src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs

```diff
@@ -148,9 +148,9 @@ public static void UpdateBarTextColor(this AToolbar nativeToolbar, Toolbar toolb
 			// We need to reset the toolbar text color to the default color when it's unset
 			if (_defaultTitleTextColor == null)
 			{
-				var a = TintTypedArray.ObtainStyledAttributes(nativeToolbar.Context?.GetThemedContext(), null, Resource.Styleable.Toolbar, Resource.Attribute.toolbarStyle, 0);
-				_defaultTitleTextColor = a.GetColorStateList(Resource.Styleable.Toolbar_titleTextColor);
-				a.Recycle();
+				var context = nativeToolbar.Context?.GetThemedContext ();
+				_defaultTitleTextColor = PlatformInterop.GetColorStateListForToolbarStyleableAttribute (context,
+					Resource.Attribute.toolbarStyle, Resource.Styleable.Toolbar_titleTextColor);
```

Comment:
@PureWeen what control should we test for this? Would we just need to test changing the toolbar color in a template using Shell?

Reply:
This behavior is currently in place for non-shell apps.   Non-shell apps only ever have a single Toolbar so the purpose of this code was to reset the `toolbar` back to the default if you navigate away to a page that has a color set and then back to one that doesn't.

So the test here would be to use a navigation page and then push to a page that changes toolbar styles or setup navigation pages inside some tabbedpages and switch between the tabs.